### PR TITLE
openmsx 17.0

### PR DESCRIPTION
**Pull request by cho-m**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

`audit` needs license.

Previous attempt was 16.0 in #64326

### CODE REVIEW

File: Formula/openmsx.rb [hunk 1]
```diff
@@ -20,31 +19,42 @@ class Openmsx < Formula
     sha256 cellar: :any, high_sierra: "86fff1a90fff96cb0398184ef7ebdeb804edda4c9d34ee5e7278159df64b10e3"
   end
 
+  depends_on "python@3.9" => :build
```
Comment (cho-m) [inline Formula/openmsx.rb:22]: I believe this is only used during build: https://openmsx.org/manual/compile.html#tools

File: Formula/openmsx.rb [hunk 2]
```diff
@@ -20,31 +19,42 @@ class Openmsx < Formula
...
+  depends_on "libvorbis"
+  depends_on "sdl2"
+  depends_on "sdl2_ttf"
+  depends_on "theora"
 
-  def install
-    # Fixes a clang crash; this is an LLVM/Apple bug, not an openmsx bug
-    # https://github.com/Homebrew/homebrew-core/pull/9753
-    # Filed with Apple: rdar://30475877
-    ENV.O0
```
Comment (cho-m) [inline Formula/openmsx.rb:33]: Might as well try removing to see if current `clang` is fine.

File: Formula/openmsx.rb [hunk 3]
```diff
@@ -20,31 +19,42 @@ class Openmsx < Formula
...
-    prefix.install Dir["derived/**/openMSX.app"]
-    bin.write_exec_script "#{prefix}/openMSX.app/Contents/MacOS/openmsx"
+
+    on_macos do
+      prefix.install Dir["derived/**/openMSX.app"]
+      bin.write_exec_script "#{prefix}/openMSX.app/Contents/MacOS/openmsx"
+    end
+    on_linux do
+      system "make", "install"
+    end
```
Comment (cho-m) [inline Formula/openmsx.rb:57]: I doubt Linux will build correctly. I just added some lines from: https://openmsx.org/manual/compile.html#installlocal

File: Formula/openmsx.rb [hunk 4]
```diff
@@ -20,31 +19,42 @@ class Openmsx < Formula
...
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "alsa-lib"
+  end
 
+  def install
     # Hardcode prefix
     inreplace "build/custom.mk", "/opt/openMSX", prefix
+    inreplace "build/probe.py", "/usr/local", HOMEBREW_PREFIX
```
Comment (cho-m) [inline Formula/openmsx.rb:41]: Not sure if there is an alternative for non-default prefix: https://github.com/openMSX/openMSX/blob/RELEASE_17_0/build/probe.py#L368

Another option I found still required `inreplace` but on `configure`: https://github.com/openMSX/openMSX/blob/master/configure#L32
In this case, the following might work:
```
$MAKE -f build/main.mk probe 3RDPARTY_INSTALL_DIR=#{HOMEBREW_PREFIX}
```

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**